src/rules/incognia-authentication.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @title Incognia Authentication Rule
+ * @overview Verify if the device logging in is at a trusted location.
+ * @gallery true
+ * @category marketplace
+ *
+ */
+async function incogniaAuthenticationRule(user, context, callback) {
+  const _ = require('lodash@4.17.19')
+
+  const { IncogniaAPI } = require('@incognia/api@1.0.0')
+  const { Auth0UserUpdateUtilities } = require('@auth0/rule-utilities@0.2.0')
+
+  const { INCOGNIA_CLIENT_ID, INCOGNIA_CLIENT_SECRET } = configuration;
+
+  if (!INCOGNIA_CLIENT_ID || !INCOGNIA_CLIENT_SECRET) {
+    console.log('Missing required configuration. Skipping.');
+    return callback(null, user, context);
+  }
+
+  // For this rule to be used, please set 'incognia_authentication_rule' as 'enabled' in your Auth0
+  // application metadata. This can be done in the advanced settings of your app.
+  // https://auth0.com/docs/get-started/dashboard/application-settings#application-metadata
+  const incogniaAuthenticationRule = _.get(context, 'clientMetadata.incognia_authentication_rule');
+  if (incogniaAuthenticationRule !== 'enabled') {
+    console.log('Incognia authentication rule is not enabled for this client. Skipping');
+    return callback(null, user, context);
+  }
+
+  const installationId = _.get(context, 'request.query.incognia_installation_id');
+  if (!installationId) {
+    console.log('Missing installation_id. Skipping.');
+    return callback(null, user, context);
+  }
+
+  const accountId = _.get(user, 'user_id');
+  if (!accountId) {
+    console.log('Missing user_id. Skipping.');
+    return callback(null, user, context);
+  }
+
+  let incogniaAPI;
+  if (global.incogniaAPI) {
+    incogniaAPI = global.incogniaAPI;
+  } else {
+    incogniaAPI = new IncogniaAPI({
+      clientId: INCOGNIA_CLIENT_ID,
+      clientSecret: INCOGNIA_CLIENT_SECRET
+    });
+    global.incogniaAPI = incogniaAPI;
+  }
+
+  try {
+    const loginAssessment = await incogniaAPI.registerLoginAssessment({
+      installationId: installationId,
+      accountId: accountId
```

Comment:
How is this `accountId` associated in Incognia with the user signing in?

Reply:
In the mobile app, the method `Incognia.setUserId` from Incognia SDK will be called during the callback of a successful login performed by the Auth0 SDK. Code would look like this:

```
val loginParams = hashMapOf("installation_id" to installationId)

WebAuthProvider.login(account)
        .withParameters(loginParams)
        .withScope("...")
        .withAudience("...")
        .start(this, object : Callback<Credentials, AuthenticationException> {
            override fun onFailure(exception: AuthenticationException) {
                // ...
            }

            override fun onSuccess(credentials: Credentials) {
                // Links this device to the account
                val jwt = JWT(credentials.idToken)
                Incognia.setUserId(applicationContext, jwt.subject)
            }
        })
```

Reply:
Gotcha 👍 so that happens in the app itself.